[Style] Generalize Style::LengthWrapperBase - Part 1: Replace WebCore::Length with Style::LengthWrapperBaseData

weinig opened the pull request:
#### c0b2282cf5ee913a2656fe2ceac0d0ea85cca0db
<pre>
[Style] Generalize Style::LengthWrapperBase - Part 1: Replace WebCore::Length with Style::LengthWrapperBaseData
<a href="https://bugs.webkit.org/show_bug.cgi?id=296632">https://bugs.webkit.org/show_bug.cgi?id=296632</a>

Reviewed by Antti Koivisto.

Copies the relevant code from `WebCore::Length` into a new type, `Style::LengthWrapperBaseData`,
and adopts it in `Style::LengthWrapperBase`. This will form the basis for further generalizations
without effecting other client of `WebCore::Length`.

To make this work, a few places that were depending on the `WebCore::Length` internals were
updated to use the `Style::LengthWrapperBase` functions instead (or, in the case of
`Style::ScrollMarginEdge`, moved off of `Style::LengthWrapperBase` entirely).

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/rendering/RenderScrollbarPart.cpp:
* Source/WebCore/rendering/style/StyleBoxData.cpp:
* Source/WebCore/rendering/style/StyleTransformData.cpp:
* Source/WebCore/style/values/flexbox/StyleFlexBasis.cpp:
* Source/WebCore/style/values/primitives/StyleLengthWrapper+CSSValueConversion.h:
* Source/WebCore/style/values/primitives/StyleLengthWrapper.h:
* Source/WebCore/style/values/primitives/StyleLengthWrapperData.cpp: Added.
* Source/WebCore/style/values/primitives/StyleLengthWrapperData.h: Added.
* Source/WebCore/style/values/scroll-snap/StyleScrollMargin.cpp:
* Source/WebCore/style/values/scroll-snap/StyleScrollMargin.h:
* Source/WebCore/style/values/scroll-snap/StyleScrollPadding.cpp:
* Source/WebCore/style/values/scroll-snap/StyleScrollPadding.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/298190@main">https://commits.webkit.org/298190@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c7ae137c10c9e4d774556faf32d48d754e89bff9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/114384 "44 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/34129 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/24593 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/120551 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65104 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/116273 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/34760 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/42690 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86927 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/41842 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117332 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27688 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102728 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67319 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26870 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20854 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/64237 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97064 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20969 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/123758 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/41398 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30880 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95750 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/41775 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98922 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95534 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24376 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40687 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18525 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/37439 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/41278 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/46786 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40870 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44178 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/42619 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9a176b8b9241b72187d3531ba6511553d7789bcb. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48662

weinig commented:
For reviewers, I wouldn't bother spending time reviewing `LengthWrapperBaseData` too much, as it is almost all just copy-paste of `Length` and will be getting modified quite a bit shortly (so I didn't do too much to clean it up).

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/502ced085d5f1b32fcf47df4f5f8c8616c8e6db6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48662

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c7ae137c10c9e4d774556faf32d48d754e89bff9. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48662

weinig commented on Source/WebCore/rendering/style/StyleBoxData.cpp:
```diff
@@ -26,6 +26,7 @@
 #include "RenderStyleConstants.h"
 #include "RenderStyleDifference.h"
 #include "RenderStyleInlines.h"
+#include "StylePrimitiveNumericTypes+Logging.h"
```
Yep, it is.

weinig commented on Source/WebCore/rendering/style/StyleTransformData.cpp:
```diff
@@ -24,6 +24,7 @@
 
 #include "RenderStyleInlines.h"
 #include "RenderStyleDifference.h"
+#include "StylePrimitiveNumericTypes+Logging.h"
```
Yep, it is.

weinig committed:
[Style] Generalize Style::LengthWrapperBase - Part 1: Replace WebCore::Length with Style::LengthWrapperBaseData
https://bugs.webkit.org/show_bug.cgi?id=296632

Reviewed by Antti Koivisto.

Copies the relevant code from `WebCore::Length` into a new type, `Style::LengthWrapperBaseData`,
and adopts it in `Style::LengthWrapperBase`. This will form the basis for further generalizations
without effecting other client of `WebCore::Length`.

To make this work, a few places that were depending on the `WebCore::Length` internals were
updated to use the `Style::LengthWrapperBase` functions instead (or, in the case of
`Style::ScrollMarginEdge`, moved off of `Style::LengthWrapperBase` entirely).

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/rendering/RenderScrollbarPart.cpp:
* Source/WebCore/rendering/style/StyleBoxData.cpp:
* Source/WebCore/rendering/style/StyleTransformData.cpp:
* Source/WebCore/style/values/flexbox/StyleFlexBasis.cpp:
* Source/WebCore/style/values/primitives/StyleLengthWrapper+CSSValueConversion.h:
* Source/WebCore/style/values/primitives/StyleLengthWrapper.h:
* Source/WebCore/style/values/primitives/StyleLengthWrapperData.cpp: Added.
* Source/WebCore/style/values/primitives/StyleLengthWrapperData.h: Added.
* Source/WebCore/style/values/scroll-snap/StyleScrollMargin.cpp:
* Source/WebCore/style/values/scroll-snap/StyleScrollMargin.h:
* Source/WebCore/style/values/scroll-snap/StyleScrollPadding.cpp:
* Source/WebCore/style/values/scroll-snap/StyleScrollPadding.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/298190@main
